common/src/main/java/com/wynntils/wynn/model/GuildAttackTimerModel.java

```diff
@@ -4,33 +4,84 @@
  */
 package com.wynntils.wynn.model;
 
+import com.wynntils.core.chat.RecipientType;
 import com.wynntils.core.managers.Model;
+import com.wynntils.core.objects.TimedSet;
+import com.wynntils.utils.Pair;
+import com.wynntils.wynn.event.ChatMessageReceivedEvent;
 import com.wynntils.wynn.model.scoreboard.Segment;
 import com.wynntils.wynn.model.scoreboard.guild.GuildAttackHandler;
 import com.wynntils.wynn.model.scoreboard.guild.TerritoryAttackTimer;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Optional;
+import java.util.concurrent.TimeUnit;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
 
 public class GuildAttackTimerModel extends Model {
     private static final Pattern GUILD_ATTACK_PATTERN = Pattern.compile("§b- (.+):(.+) §3(.+)");
+    private static final Pattern GUILD_DEFENSE_CHAT_PATTERN = Pattern.compile("§r§3.+§b (.+) defense is (.+)");
+
     public static final GuildAttackHandler SCOREBOARD_HANDLER = new GuildAttackHandler();
+    private static final TimedSet<Pair<String, String>> territoryDefenseSet = new TimedSet<>(5, TimeUnit.SECONDS, true);
 
     private static List<TerritoryAttackTimer> attackTimers = List.of();
 
     public static void init() {}
 
+    @SubscribeEvent
+    public static void onMessage(ChatMessageReceivedEvent event) {
+        if (event.getRecipientType() != RecipientType.GUILD) return;
+
+        Matcher matcher = GUILD_DEFENSE_CHAT_PATTERN.matcher(event.getCodedMessage());
```

Comment:
Can be faked, but we can't do much about that.

Reply:
Yeah